daemon/daemon_unix.go

```diff
@@ -550,7 +550,7 @@ func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.
 	// check for various conflicting options with user namespaces
 	if daemon.configStore.RemappedRoot != "" && hostConfig.UsernsMode.IsPrivate() {
 		if hostConfig.Privileged {
-			return warnings, fmt.Errorf("Privileged mode is incompatible with user namespaces")
+			return warnings, fmt.Errorf("Privileged mode is incompatible with user namespaces.  To temporarily disable user namespaces, use '--userns=host'.")
```

Comment:
Perhaps the equivalent of `--userns=host` in the API should be used, but not sure if that's equally useful for the situation you're trying to resolve